src/requests_utilities.py

```diff
@@ -55,33 +66,39 @@ def __assert_status_code(self):
         )
         logger.info("Status is %s", self.status_code)
 
-    def get(
+    def __make_request(
         self,
-        endpoint: str,
-        headers: dict | None = None,
-        expected_status_code=200,
+        method: str,
+        request_params: RequestParams,
+        auth_headers: dict | None = None,
     ):
         """
-        Perform a GET request to the specified API endpoint.
+        Perform an HTTP request to the specified API endpoint.
         """
 
-        logger.info("Starting GET method.")
+        logger.info("Starting %s method.", method.upper())
 
-        if not headers:
-            headers = {"Content-Type": "application/json"}
+        if not auth_headers:
+            auth_headers = {"Content-Type": "application/json"}
         else:
-            headers.update({"Content-Type": "application/json"})
+            auth_headers.update({"Content-Type": "application/json"})
+
+        if request_params.auth_extra:
+            auth_headers.update(request_params.auth_extra)
 
-        self.url = self.base_url + endpoint
+        self.url = self.get_base_url() + request_params.endpoint
         logger.info("URL: %s", self.url)
 
-        self.expected_status_code = expected_status_code
+        self.expected_status_code = request_params.expected_status_code
 
-        self.response_api = requests.get(
+        self.response_api = requests.request(
+            method=method,
             url=self.url,
-            headers=headers,
-            timeout=5,
+            json=request_params.payload,
+            headers=auth_headers,
+            timeout=10,
         )
+
         self.status_code = self.response_api.status_code
         self.__assert_status_code()
```

Comment:
You are putting a lot of information in the class instance, `self.status_code` is going to use only 1 time

```suggestion
        status_code = self.response_api.status_code
        self.__assert_status_code(status_code, request_params.expected_status_code)
```

Reply:
`information in the class instance` fixed
commit: 27d269d0